packages/core/types/src/http/customer/common.ts

```diff
@@ -0,0 +1,97 @@
+import { BaseFilterable, OperatorMap } from "../../dal"
+
+export interface BaseCustomerGroup {
+  id: string
+  name: string | null
+  customers: BaseCustomer[]
+  metadata: Record<string, unknown> | null
+  created_at: string
+  updated_at: string
+}
+
+export interface BaseCustomerAddress {
+  id: string
+  address_name: string | null
+  is_default_shipping: boolean
+  is_default_billing: boolean
+  customer_id: string
+  company: string | null
+  first_name: string | null
+  last_name: string | null
+  address_1: string | null
+  address_2: string | null
+  city: string | null
+  country_code: string | null
+  province: string | null
+  postal_code: string | null
+  phone: string | null
+  metadata: Record<string, unknown> | null
+  created_at: string
+  updated_at: string
```

Comment:
Q: isn't it always Date | string?

Reply:
@adrien2p I think when sending to the BE it first needs to be serialized to string, and the response will also need to be parsed from string to Date, that doesn't happen automatically currently.